### PR TITLE
feat: wire-up THORCHain pools

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

- Fetches pools and user data from THOR API and massages it.

Note for unclaimed fees, TVL, 24h and 7d volume, there may be some follow-ups needed to improve the math WRT:

- rebalancing of the pool since deposit
- using the same 24h window as THORYield

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

- contributes to https://github.com/shapeshift/web/issues/5939 , though rebalanced data will still need to be tackled as a follow-up

## Risk

N/A

## Testing

- Data looks sane against THORYield / THORSWap

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

### This diff

image
image

### THORYield

image
image
image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/5978?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `develop`
  * **PR #5966** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5966?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5969** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5969?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5974** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5974?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5978** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5978?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #5970** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5970?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5978?utm_source=stack-comment).

File: src/pages/ThorChainLP/YourPositions.tsx [hunk 1]
```diff
@@ -101,22 +137,32 @@ export const YourPositions = () => {
...
-      )
+      // TODO(gomes): add a custom query hook for this and useQueries()
+      // this is a temporary way to achieve this
+      return parsedPools.map(pool => {
+        return (
+          <PositionButton
+            assetId={pool.assetId}
+            name={pool.name}
+            opportunityId={pool.opportunityId}
+            apy={pool.poolAPY}
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/YourPositions.tsx:152]: May or may not be the correct one to use https://gitlab.com/thorchain/midgard/-/merge_requests/564

**Review by 0xApotheosis — APPROVED** [review]
I'm correctly seeing my ERC20 asset pools, though not my BCH pool for some reason. I'm happy to get this in in the name of progress and I'll investigate in a follow-up PR!

Also, the console logs are pretty wild atm, but I'll investigate in a follow up PR.

File: src/lib/utils/thorchain/lp.ts [hunk 2]
```diff
@@ -50,13 +51,24 @@ export const getAllThorchainLiquidityMembers = async (): Promise<MidgardLiquidit
...
+  try {
+    const { data } = await axios.get<MidgardLiquidityProvider>(
+      `${getConfig().REACT_APP_MIDGARD_URL}/member/${address}`,
+    )
+
+    return data
+  } catch (e) {
+    // THORCHain returns a 404 which is perfectly valid, but axios catches as an error
+    // We only want to log errors to the console if they're actual errors, not 404s
+    if ((e as AxiosError).isAxiosError && (e as AxiosError).response?.status !== 404)
```
[the remark this answers is not in the document]
Comment (0xApotheosis) [reply src/lib/utils/thorchain/lp.ts:67]: Oh nevermind, these aren't from the console log.

**Pull request merged**